From ff48e72ac1ba945a6a37826eaa8f7667d51ab758 Mon Sep 17 00:00:00 2001
From: "Unnithan, Balakrishnan" <balakrishnan.unnithan@intel.com>
Date: Sat, 7 Dec 2024 17:10:41 +0530
Subject: [PATCH 7/7] Configure virtual_ab compression

Set vabc algo to lz4.
Enabled vabc compression threads.
Enabled vabc compression xor.

Tests done: Build-Flash-Verify

Tracked-on: OAM-125218
Signed-off-by: Unnithan, Balakrishnan <balakrishnan.unnithan@intel.com>
---
 target/product/virtual_ab_ota/vabc_features.mk | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/target/product/virtual_ab_ota/vabc_features.mk b/target/product/virtual_ab_ota/vabc_features.mk
index 1219763b0d..7b90106fbd 100644
--- a/target/product/virtual_ab_ota/vabc_features.mk
+++ b/target/product/virtual_ab_ota/vabc_features.mk
@@ -52,13 +52,16 @@ PRODUCT_VENDOR_PROPERTIES += ro.virtual_ab.batch_writes=true
 
 # Enabling this property, will improve OTA install time
 # but will use an additional CPU core
-# PRODUCT_VENDOR_PROPERTIES += ro.virtual_ab.compression.threads=true
-ifndef PRODUCT_VIRTUAL_AB_COMPRESSION_FACTOR
-    PRODUCT_VIRTUAL_AB_COMPRESSION_FACTOR := 65536
-endif
+PRODUCT_VENDOR_PROPERTIES += ro.virtual_ab.compression.threads=true
 
 PRODUCT_VIRTUAL_AB_COMPRESSION := true
-PRODUCT_VIRTUAL_AB_COMPRESSION_METHOD ?= none
+PRODUCT_VIRTUAL_AB_COMPRESSION_METHOD ?= lz4
+PRODUCT_VIRTUAL_AB_COMPRESSION_FACTOR ?= 65536
+
 PRODUCT_PACKAGES += \
+    snapuserd.vendor_ramdisk \
     snapuserd \
+    snapuserd.recovery
+
+$(call inherit-product, $(SRC_TARGET_DIR)/product/virtual_ab_ota/launch_with_vendor_ramdisk.mk)
 
-- 
2.47.1

